packages/turbo/src/stradaBridge.ts

```diff
@@ -0,0 +1,223 @@
+    }
+
+    // Native handler
+
+    postMessage(message) {
+      ${
+        Platform.OS === 'android'
+          ? 'StradaNative.postMessage(JSON.stringify(message))'
+          : 'webkit.messageHandlers.stradaNative.postMessage(message)'
+      }
+    }
+
+    // Web global
+
+    get isStradaAvailable() {
+      return window.Strada;
+    }
+
+    get webBridge() {
+      return window.Strada.web;
+    }
+  }
+
+  if (
+    document.readyState === 'interactive' ||
+    document.readyState === 'complete'
+  ) {
+    initializeBridge();
+  } else {
+    document.addEventListener('DOMContentLoaded', () => {
+      initializeBridge();
+    });
+  }
+
+  function initializeBridge() {
+    window.nativeBridge = new NativeBridge();
+  }
+})();
+`;
+
+const useStradaBridge = (
+  url: string,
+  sessionHandle: string,
+  stradaComponents?: StradaComponent[]
+) => {
+  const initializeStradaBridge = useCallback(async () => {
+    const isStradaUndefined = await RNVisitableViewModule.injectJavaScript(
+      sessionHandle,
+      `window.nativeBridge == undefined`
+    );
+    if (isStradaUndefined) {
+      await RNVisitableViewModule.injectJavaScript(
+        sessionHandle,
+        stradaBridgeScript
+      );
+      if (!stradaComponents) {
+        return;
+      }
+      Promise.all(
+        stradaComponents.map(({ name }) =>
+          RNVisitableViewModule.injectJavaScript(
+            sessionHandle,
+            `window.nativeBridge.register('${name}')`
+          )
+        )
+      );
+    }
+  }, [sessionHandle, stradaComponents]);
+
+  const stradaUserAgent = useMemo(
+    () =>
+      stradaComponents
+        ? ` bridge-components: [${stradaComponents.reduce(
+            (acc, { name }) => `${acc} ${name}`,
+            ''
+          )} ]`
+        : '',
+    [stradaComponents]
+  );
+
+  const useStrada = (name: string, onReceive: any) => {
+    const previousMessages = useRef<{ [event: string]: StradaEvent }>({});
+
+    const didReceive = useCallback(
+      (e) => {
+        if (url !== (e as StradaEvent)?.data.metadata.url) {
+          return;
+        }
+        previousMessages.current[e.event] = e;
+        onReceive(e);
+      },
+      [onReceive]
+    );
+
+    useEffect(() => {
+      const eventHandlerEventSubscription = registerMessageEventListener(
+        name,
+        didReceive
+      );
+      return () => {
+        eventHandlerEventSubscription.remove();
+      };
+    }, [name, onReceive, didReceive]);
+
+    return useCallback(
+      (event: string, data?: object) => {
+        const previousMessage = previousMessages.current[event];
+        const messageToSend = previousMessage
+          ? {
+              ...previousMessage,
+              data: { ...previousMessage.data, ...data },
+            }
+          : {
+              component: name,
+              event,
+              data: {
+                metadata: {
+                  url,
+                },
+                ...data,
```

Comment:
i believe you should spread `data` before setting `metadata` (otherwise your metadata can be overridden by data.metadata)